Bottom: 0386ea113fd71e596914df1d4107ebea6c72b280
Top:    bdef5cd2b170f152c96af349f59d0e97bcb23b63
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-29 09:49:57 -0500

Refresh of fixup-asset-host

---
diff --git a/docker/buildbot-worker-ptxdist/buildbot/.pydistutils.cfg b/docker/buildbot-worker-ptxdist/buildbot/.pydistutils.cfg
deleted file mode 100644
index ee683c3d..00000000
--- a/docker/buildbot-worker-ptxdist/buildbot/.pydistutils.cfg
+++ /dev/null
@@ -1,2 +0,0 @@
-[easy_install]
-index-url=http://nexus.novatech-llc.com/repository/pypi/simple
diff --git a/roles/buildbot-master/files/openembedded.py b/roles/buildbot-master/files/openembedded.py
index 9166ece0..042d283b 100755
--- a/roles/buildbot-master/files/openembedded.py
+++ b/roles/buildbot-master/files/openembedded.py
@@ -9,7 +9,7 @@ c = WorkerConfig = {}
 
 
 DEFAULT_BBFLAGS = '-k'
-DEFAULT_CODEBASE = "ntel"
+ASSET_HOST = os.getenv("ASSET_HOST", default="http://127.0.0.1")
 
 # Workers
 # The 'workers' list defines the set of recognized buildworkers. Each element is
diff --git a/roles/buildbot-worker-ntel/defaults/main.yml b/roles/buildbot-worker-ntel/defaults/main.yml
index 8caf5b7b..0f440dc6 100644
--- a/roles/buildbot-worker-ntel/defaults/main.yml
+++ b/roles/buildbot-worker-ntel/defaults/main.yml
@@ -54,6 +54,9 @@ container_volumes:
 - '{{ data_volume }}:{{ data_path }}:z'
 - '{{ cache_volume }}:{{ cache_path }}:z'
 - '{{ secrets_hostdir }}:{{ secrets_path }}'
+- '{{ config_hostdir }}/pydistutils.cfg:/home/buildbot/.pydistutils.cfg'
+- '{{ config_hostdir }}/ptxdistrc:/home/buildbot/.ptxdist/ptxdistrc-2012.09'
+- '{{ config_hostdir }}/netrc:/home/buildbot/.netrc'
 
 # mount path of the data volume
 data_path: '/buildbot'
diff --git a/roles/buildbot-worker-ntel/tasks/main.yml b/roles/buildbot-worker-ntel/tasks/main.yml
index d96081aa..7c865e5c 100644
--- a/roles/buildbot-worker-ntel/tasks/main.yml
+++ b/roles/buildbot-worker-ntel/tasks/main.yml
@@ -25,7 +25,7 @@
     - '{{ cache_volume }}:{{ cache_path }}:z'
     detach: no
     cleanup: yes
-    command: /bin/ash -c 
+    command: /bin/ash -c
       "mkdir -p {{item}} &&
       chown -c -R {{buildbot_uid}}:{{buildbot_uid}} {{item}} &&
       chmod -v 777 {{item}}"
@@ -44,12 +44,24 @@
     - '{{ data_volume }}:{{ data_path }}:z'
     detach: no
     cleanup: yes
-    command: /bin/ash -c 
+    command: /bin/ash -c
       "mkdir -p {{item}} &&
       if [ $(stat --printf='%u' '{{item}}') -ne {{buildbot_uid}} ]; then
         chown -c -R {{buildbot_uid}}:{{buildbot_uid}} {{item}};
       fi"
 
+- name: Create configuration directory
+  file:
+    path: '{{ config_hostdir }}'
+    state: directory
+
+- name: Generate distutils configuration
+  template:
+    src: pydistutils.cfg.j2
+    dest: '{{ config_hostdir }}/pydistutils.cfg'
+    group: '{{ buildbot_uid }}'
+    owner: '{{ buildbot_uid }}'
+
 - name: Populate assets secret
   template:
     src: "netrc.j2"
@@ -75,6 +87,7 @@
     repo: '{{ image_repo }}'
     version: master
     dest: '{{ image_dir }}'
+    force: yes
 
 - name: Create buildbot worker image
   docker_image:
diff --git a/roles/buildbot-worker-ntel/templates/netrc.j2 b/roles/buildbot-worker-ntel/templates/netrc.j2
new file mode 100644
index 00000000..e81d1062
--- /dev/null
+++ b/roles/buildbot-worker-ntel/templates/netrc.j2
@@ -0,0 +1,3 @@
+machine {{ asset_host | urlsplit('hostname') }}
+login {{ asset_user }}
+password {{ asset_password }}
diff --git a/roles/buildbot-worker-ntel/templates/ptxdistrc.j2 b/roles/buildbot-worker-ntel/templates/ptxdistrc.j2
new file mode 100644
index 00000000..4d2b48e9
--- /dev/null
+++ b/roles/buildbot-worker-ntel/templates/ptxdistrc.j2
@@ -0,0 +1,76 @@
+#
+# Automatically generated file; DO NOT EDIT.
+# PTXdist 2012.09.1-00373-gee9622f
+#
+
+#
+# PTXDIST Setup Menu
+#
+
+#
+# User
+#
+PTXCONF_SETUP_USER_NAME="NovaTech Buildbot"
+PTXCONF_SETUP_USER_EMAIL="buildbot@novatechweb.com"
+
+#
+# Proxies
+#
+PTXCONF_SETUP_FTP_PROXY=""
+PTXCONF_SETUP_HTTP_PROXY=""
+PTXCONF_SETUP_HTTPS_PROXY=""
+PTXCONF_SETUP_NO_PROXY=""
+
+#
+# Project Searchpath
+#
+PTXCONF_SETUP_PROJECTPATH="${PTXDIST_TOPDIR}/projects"
+
+#
+# Source Directories
+#
+PTXCONF_SETUP_SRCDIR="{{cache_path}}/downloads"
+
+#
+# Source Download
+#
+# PTXCONF_SETUP_NO_DOWNLOAD is not set
+# PTXCONF_SETUP_PTXMIRROR_ONLY is not set
+PTXCONF_SETUP_PTXMIRROR="{{asset_host}}/orphans"
+PTXCONF_SETUP_DEBMIRROR="{{asset_host}}/debian"
+PTXCONF_SETUP_SFMIRROR="{{asset_host}}/sourceforge"
+PTXCONF_SETUP_GNUMIRROR="{{asset_host}}/gnu"
+PTXCONF_SETUP_XORGMIRROR="{{asset_host}}/xorg"
+PTXCONF_SETUP_KERNELMIRROR="{{asset_host}}/kernel"
+PTXCONF_SETUP_PYPIMIRROR="{{asset_host}}/pypi"
+# PTXCONF_SETUP_CHECK_ALWAYS is not set
+PTXCONF_SETUP_CHECK_NOTEMPTY=y
+# PTXCONF_SETUP_CHECK_NEVER is not set
+PTXCONF_SETUP_CHECK="notempty"
+
+#
+# IPKG Repository
+#
+PTXCONF_SETUP_IPKG_REPOSITORY="{{cache_path}}/ipkg-repository"
+
+#
+# Java SDK
+#
+PTXCONF_SETUP_JAVA_SDK="/usr"
+
+#
+# Developer Options
+#
+# PTXCONF_SETUP_DISABLE_LOCAL_CHECK is not set
+PTXCONF_SETUP_ENV_WHITELIST=""
+# PTXCONF_SETUP_COMMON_CACHE is not set
+# PTXCONF_SETUP_GEN_DEP_TREE is not set
+# PTXCONF_SETUP_CHECK_EXIT_ON_ERROR is not set
+PTXCONF_SETUP_CCACHE=y
+PTXCONF_SETUP_PATCHIN_GIT=y
+# PTXCONF_SETUP_NFS_REL_SYMLINK is not set
+# PTXCONF_SETUP_DIRECT_CLEAN is not set
+PTXCONF_SETUP_HOST_CPP="cpp"
+PTXCONF_SETUP_HOST_CC="gcc"
+PTXCONF_SETUP_HOST_CXX="g++"
+PTXCONF_SETUP_HOST_MAKE="make"
diff --git a/roles/buildbot-worker-ntel/templates/pydistutils.cfg.j2 b/roles/buildbot-worker-ntel/templates/pydistutils.cfg.j2
new file mode 100644
index 00000000..1d205ea2
--- /dev/null
+++ b/roles/buildbot-worker-ntel/templates/pydistutils.cfg.j2
@@ -0,0 +1,2 @@
+[easy_install]
+index_url = {{asset_host}}/pypi/simple
